test/jdk/sun/security/tools/keytool/i18n.java

```diff
@@ -23,17 +23,410 @@
+                    + "-addprovider SUN",
+                    "Output in ${LANG}. Check: contains 2 keystore entries "
+                            + "(alias name mykey & mykey2)."},
+
+            {"-storepasswd "
+                    + "-storepass password "
+                    + "-new a "
+                    + "-keystore ./i18n.keystore",
+                    "Output in ${LANG}. Check keytool error: java.lang.Exception: "
+                            + "New password must be at least 6 characters."},
+
+            {"-storepasswd "
+                    + "-storetype PKCS11 "
+                    + "-keystore NONE",
+                    "Output in ${LANG}. Check keytool error: java.lang"
+                            + ".UnsupportedOperationException: -storepasswd and "
+                            + "-keypasswd commands not supported if -storetype is"
+                            + " PKCS11."},
+
+            {"-keypasswd "
+                    + "-storetype PKCS11 "
+                    + "-keystore NONE",
+                    "Output in ${LANG}. Check keytool error: java.lang"
+                            + ".UnsupportedOperationException: -storepasswd and "
+                            + "-keypasswd commands not supported if -storetype is"
+                            + " PKCS11."},
+
+            {"-list -protected "
+                    + "-storepass password "
+                    + "-keystore ./i18n.keystore",
+                    "Output in ${LANG}. Check keytool error: java.lang"
+                            + ".IllegalArgumentException: if -protected is "
+                            + "specified, then -storepass, -keypass, and -new "
+                            + "must not be specified."},
+
+            {"-keypasswd -protected "
+                    + "-storepass password "
+                    + "-keystore ./i18n.keystore",
+                    "Output in ${LANG}. Check keytool error: java.lang"
+                            + ".IllegalArgumentException: if -protected is "
+                            + "specified, then -storepass, -keypass, and -new "
+                            + "must not be specified."},
+
+            {"-keypasswd -protected "
+                    + "-storepass password "
+                    + "-new aaaaaa "
+                    + "-keystore ./i18n.keystore",
+                    "Output in ${LANG}. Check keytool error: java.lang"
+                            + ".IllegalArgumentException: if -protected is "
+                            + "specified, then -storepass, -keypass, and -new "
+                            + "must not be specified."},
+    };
+    private static String TEST_SRC = System.getProperty("test.src");
+    private static int TIMEOUT_MS = 120000;
+    private volatile boolean failed = false;
+    private volatile boolean aborted = false;
+    private Thread currentThread = null;
+
+    public static class DialogBuilder {
```

Comment:
This seems to be a boilerplate for displaying the panel. Could this be separated from the test and converted into some library?

Reply:
Sure, we can move this code to the library as there is no test code in it.